codegen/templates/endpoint.tmpl

```diff
@@ -146,11 +181,19 @@ func (w {{$workflow}}) Handle(
 		return {{if eq .ResponseType ""}}nil, err{{else}}nil, nil, err{{end}}
 	}
 
+	// Filter and map response headers from client to server response.
+	endRespHead := map[string]string{}
+	{{if (ne (len $resHeaderMap) 0) -}}
+	for k, v := range {{$resHeaderMap | printf "%#v" }} {
+		endRespHead[v] = respHeaders[k]
```

Comment:
I'm not sure that's less confusing. The map is taking header X and putting it in header Y. On the request path that server->client, on the response that's client->server.